routing-tutorial-app/src/main/java/com/esri/arcgismaps/sample/routingtutorialapp/screens/MainScreen.kt

```diff
@@ -0,0 +1,458 @@
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.PaddingValues
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.width
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.text.KeyboardActions
+import androidx.compose.foundation.text.KeyboardOptions
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.LocationOn
+import androidx.compose.material.icons.outlined.LocationOn
+import androidx.compose.material.icons.outlined.Refresh
+import androidx.compose.material.icons.outlined.Search
+import androidx.compose.material3.BottomSheetScaffold
+import androidx.compose.material3.Button
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.material3.FloatingActionButton
+import androidx.compose.material3.HorizontalDivider
+import androidx.compose.material3.Icon
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.SnackbarHostState
+import androidx.compose.material3.Text
+import androidx.compose.material3.TextField
+import androidx.compose.material3.rememberBottomSheetScaffoldState
+import androidx.compose.material3.rememberStandardBottomSheetState
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.rememberCoroutineScope
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.LocalConfiguration
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.platform.LocalFocusManager
+import androidx.compose.ui.platform.LocalSoftwareKeyboardController
+import androidx.compose.ui.text.font.FontStyle
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.input.ImeAction
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.dp
+import com.arcgismaps.mapping.view.LocationDisplay
+import com.arcgismaps.toolkit.geoviewcompose.MapView
+import com.arcgismaps.toolkit.geoviewcompose.rememberLocationDisplay
+import com.esri.arcgismaps.sample.routingtutorialapp.components.MapViewModel
+import com.esri.arcgismaps.sample.sampleslib.components.LoadingDialog
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.launch
+
+/**
+ * Main screen layout for the sample app
+ */
+
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun MainScreen(sampleName: String) {
+
+    // Create and remember a location display with a recenter auto pan mode.
+    val coroutineScope = rememberCoroutineScope()
+    val locationDisplay = rememberLocationDisplay()
```

Comment:
Ah when I do this, for some reason everytime I click on the find route to find an address between the two routes, it pans back to user's location instead of panning to show the route and the markers.

Reply:
Nevermind I believe I found a way to resolve it by assigning currentJob to panning the route and stops

Reply:
@NourdotSiwar  Not sure if I can repro this, I am able to solve the route with a visible route with markers. Did you remove the usage of `locationDisplay` from the ViewModel?

Reply:
i do get the routes and markers, the only issue is when I solve the route, it pans to the user's location instead of showing the whole route and markers. I also have removed locationDisplay from viewmodel